src/include/detail/ivf/index.h

```diff
@@ -109,16 +113,18 @@ int ivf_index(
     // be difficult to implement.  Even this algorithm is not trivial to
     // parallelize, because of the random access to the indices array.
     for (size_t i = 0; i < db.num_cols(); ++i) {
-      size_t bin = parts[i];
-      size_t ibin = indices[bin];
+      if (auto iter = deleted_ids_set.find(external_ids[i]); iter == deleted_ids_set.end()) {
```

Comment:
Same question about `==`

Reply:
This is filtering the vectors that are deleted and should be skipped.